59.md

```diff
@@ -0,0 +1,266 @@
+NIP-59
+======
+
+Gift Wrap
+---------
+
+`draft` `optional` `author:vitorpamplona` `author:kieran` `author:paulmillr` `author:staab`
+
+This NIP defines a protocol for encapsulating any nostr event. This makes it possible to obscure almost all metadata for a given event, perform collaborative signing, and more.
+
+This NIP relies on [NIP-44](44.md)'s versioned encryption algorithms.
+
+# Overview
+
+This protocol uses three main concepts to protect the transmission of a target event: `rumor`s, `seal`s, and `gift wrap`s.
+
+- A `rumor` is a regular nostr event, but is **not signed**. This means that if it is leaked, it cannot be verified.
```

Comment:
I'm skeptical about calling this a rumor. Even though the name perfectly describes it, it doesn't fit the paradigm of `EventTemplate` -> `UnsignedEvent` -> `Event`.

https://github.com/nbd-wtf/nostr-tools/blob/master/event.ts#L45

I think it is most close to `UnsignedEvent`, except that in UnsignedEvent the `id` is optional.

`rumor` is also shorter and has a specific connotation that `UnsignedEvent` doesn't. So I have mixed feelings.

Reply:
My library nostr-types takes a `PreEvent` into an `Event` by computing an `Id` and adding a `Signature`.   A `Rumor` is half-way between those two concepts.  But I just did the pragmatic thing and made a new type, and the code that turns a PreEvent to an Event doesn't pass it through the Rumor stage.  Does it look like it could be refactored. Sure. But I'm not sure.

Reply:
>  it doesn't fit the paradigm of EventTemplate -> UnsignedEvent -> Event.

Seems to me a good argument for a new term. Although arguably according to the above paradigm it should be something more like `EventTemplate` -> `RawEvent` -> `UnsignedEvent` -> `Event`.